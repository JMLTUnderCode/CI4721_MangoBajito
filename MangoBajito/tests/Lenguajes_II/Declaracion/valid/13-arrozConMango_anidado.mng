# Test 13: Definicion y uso de estructuras anidadas.
se_prende(){
	arroz_con_mango Persona {
		nombre : higuerote;
		sexo : negro;
		edad : mango;
	};

	arroz_con_mango Registro {
		codigo : mango;
		usuario : Persona;
	};

	#culito user : Persona = {"Junior", 'M', 26};
	culito registro_1 : Registro = {0, {"Junior", 'M', 26}};

	#culito registro_2 : Registro;
	#registro_2 = {1, {"Astrid", 'F', 23}};

	#culito registro_3 : Registro;
	#registro_3.codigo = 2;
	#registro_3.usuario = {"Jhonaiker", 'M', 25}};
};