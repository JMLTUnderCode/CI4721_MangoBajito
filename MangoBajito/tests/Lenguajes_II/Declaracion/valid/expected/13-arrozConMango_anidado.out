| AcomÃ³date que te lo estan analizando d:-)
--> Print table? (s/n): [1;36m[5m
               =======================================================
                             Symbol Table Representation              
               =======================================================
[0m
[1;35m[5m    Clave: 1: registro_1.usuario.edad[0m
    |---> SÃ­mbolo: registro_1.usuario.edad, CategorÃ­a: (10)STRUCT_ATTRIBUTE, Scope: 2, Type: mango, Value: 26

[1;35m[5m    Clave: 2: registro_1.usuario.nombre[0m
    |---> SÃ­mbolo: registro_1.usuario.nombre, CategorÃ­a: (10)STRUCT_ATTRIBUTE, Scope: 2, Type: higuerote, Value: "Junior"

[1;35m[5m    Clave: 3: registro_1.codigo[0m
    |---> SÃ­mbolo: registro_1.codigo, CategorÃ­a: (10)STRUCT_ATTRIBUTE, Scope: 2, Type: mango, Value: 0

[1;35m[5m    Clave: 4: Registro.usuario[0m
    |---> SÃ­mbolo: Registro.usuario, CategorÃ­a: (10)STRUCT_ATTRIBUTE, Scope: 4, Type: Persona

[1;35m[5m    Clave: 5: Registro.codigo[0m
    |---> SÃ­mbolo: Registro.codigo, CategorÃ­a: (10)STRUCT_ATTRIBUTE, Scope: 4, Type: mango

[1;35m[5m    Clave: 6: Registro[0m
    |---> SÃ­mbolo: Registro, CategorÃ­a: (4)STRUCT, Scope: 2, Informacion: Registro.codigo Registro.usuario 

[1;35m[5m    Clave: 7: Persona.edad[0m
    |---> SÃ­mbolo: Persona.edad, CategorÃ­a: (10)STRUCT_ATTRIBUTE, Scope: 3, Type: mango

[1;35m[5m    Clave: 8: mango[0m
    |---> SÃ­mbolo: mango, CategorÃ­a: (7)TYPE, Scope: 0

[1;35m[5m    Clave: 9: manguita[0m
    |---> SÃ­mbolo: manguita, CategorÃ­a: (7)TYPE, Scope: 0

[1;35m[5m    Clave: 10: manguangua[0m
    |---> SÃ­mbolo: manguangua, CategorÃ­a: (7)TYPE, Scope: 0

[1;35m[5m    Clave: 11: Persona.nombre[0m
    |---> SÃ­mbolo: Persona.nombre, CategorÃ­a: (10)STRUCT_ATTRIBUTE, Scope: 3, Type: higuerote

[1;35m[5m    Clave: 12: Persona[0m
    |---> SÃ­mbolo: Persona, CategorÃ­a: (4)STRUCT, Scope: 2, Informacion: Persona.nombre Persona.sexo Persona.edad 

[1;35m[5m    Clave: 13: negro[0m
    |---> SÃ­mbolo: negro, CategorÃ­a: (7)TYPE, Scope: 0

[1;35m[5m    Clave: 14: registro_1.usuario.sexo[0m
    |---> SÃ­mbolo: registro_1.usuario.sexo, CategorÃ­a: (10)STRUCT_ATTRIBUTE, Scope: 2, Type: negro, Value: 'M'

[1;35m[5m    Clave: 15: tas_claro[0m
    |---> SÃ­mbolo: tas_claro, CategorÃ­a: (7)TYPE, Scope: 0

[1;35m[5m    Clave: 16: registro_1.usuario[0m
    |---> SÃ­mbolo: registro_1.usuario, CategorÃ­a: (4)STRUCT, Scope: 2, Type: Persona, Informacion: registro_1.usuario.nombre registro_1.usuario.sexo registro_1.usuario.edad 

[1;35m[5m    Clave: 17: hablame[0m
    |---> SÃ­mbolo: hablame, CategorÃ­a: (2)FUNCTION, Scope: 0

[1;35m[5m    Clave: 18: registro_1[0m
    |---> SÃ­mbolo: registro_1, CategorÃ­a: (4)STRUCT, Scope: 2, Type: Registro, Informacion: registro_1.codigo registro_1.usuario 

[1;35m[5m    Clave: 19: un_coÃ±o[0m
    |---> SÃ­mbolo: un_coÃ±o, CategorÃ­a: (7)TYPE, Scope: 0

[1;35m[5m    Clave: 20: funcion$[0m
    |---> SÃ­mbolo: funcion$, CategorÃ­a: (7)TYPE, Scope: 0

[1;35m[5m    Clave: 21: Persona.sexo[0m
    |---> SÃ­mbolo: Persona.sexo, CategorÃ­a: (10)STRUCT_ATTRIBUTE, Scope: 3, Type: negro

[1;35m[5m    Clave: 22: caramba_Ã±ero[0m
    |---> SÃ­mbolo: caramba_Ã±ero, CategorÃ­a: (7)TYPE, Scope: 0

[1;35m[5m    Clave: 23: se_prende[0m
    |---> SÃ­mbolo: se_prende, CategorÃ­a: (2)FUNCTION, Scope: 0

[1;35m[5m    Clave: 24: higuerote[0m
    |---> SÃ­mbolo: higuerote, CategorÃ­a: (7)TYPE, Scope: 0

[1;35m[5m    Clave: 25: rescata[0m
    |---> SÃ­mbolo: rescata, CategorÃ­a: (2)FUNCTION, Scope: 0

--> Print AST? (s/n): [1;36m[5m
               =======================================================
                                   AST Representation                 
               =======================================================
[0m
Programa
|--> Main
     |--> InstrucciÃ³n | Kind: ;
          |--> InstrucciÃ³n | Kind: ;
          |    |--> Persona | Category: DeclaraciÃ³n | Type: arroz_con_mango | Kind: Estructura
          |    |    |--> InstrucciÃ³n | Kind: ;
          |    |         |--> InstrucciÃ³n | Kind: ;
          |    |         |    |--> nombre | Category: Atributo_Estructura | Type: higuerote | Value: ""
          |    |         |    |--> sexo | Category: Atributo_Estructura | Type: negro | Value: ''
          |    |         |--> edad | Category: Atributo_Estructura | Type: mango | Value: 0
          |    |--> Registro | Category: DeclaraciÃ³n | Type: arroz_con_mango | Kind: Estructura
          |         |--> InstrucciÃ³n | Kind: ;
          |              |--> codigo | Category: Atributo_Estructura | Type: mango | Value: 0
          |              |--> usuario | Category: Atributo_Estructura | Type: Persona
          |--> AsignaciÃ³n | Kind: =
               |--> registro_1 | Category: DeclaraciÃ³n | Type: Registro | Kind: VARIABLE
               |--> Literal | Category: Estructura
                    |--> Secuencia | Category: ExpresiÃ³n | Kind: ,
                         |--> Literal | Category: NumÃ©rico | Type: mango | Value: 0
                         |--> Literal | Category: Estructura
                              |--> Secuencia | Category: ExpresiÃ³n | Kind: ,
                                   |--> Secuencia | Category: ExpresiÃ³n | Kind: ,
                                   |    |--> Literal | Category: Cadena de Caracteres | Type: higuerote | Value: "Junior"
                                   |    |--> Literal | Category: Caracter | Type: negro | Value: 'M'
                                   |--> Literal | Category: NumÃ©rico | Type: mango | Value: 26
--> Print TAC? (s/n): [1;32m[5m
               =======================================================
                       ---->       Correct Program       <----        
               =======================================================
[0m
| Hay un poquito de cÃ³digo en tu marihuana d:-]
